src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -294,6 +294,45 @@ public void GetNullValue()
             Assert.Null(config.GetSection("Object").Get<ComplexOptions>());
         }
 
+        [Fact]
+        public void ThrowsIfPropertyInConfigMissingInModel()
+        {
+            var dic = new Dictionary<string, string>
+            {
+                {"ThisDoesNotExistInTheModel", "42"},
+                {"Integer", "-2"},
+                {"Boolean", "TRUe"},
+                {"Nested:Integer", "11"}
+            };
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dic);
+            var config = configurationBuilder.Build();
+
+            var instance = new ComplexOptions();
+
+            Assert.Throws<InvalidOperationException>(
+                () => config.Bind(instance, o => o.ErrorOnUnknownConfiguration = true));
+        }
+        [Fact]
+        public void ThrowsIfPropertyInConfigMissingInNestedModel()
+        {
+            var dic = new Dictionary<string, string>
+            {
+                {"Nested:ThisDoesNotExistInTheModel", "42"},
+                {"Integer", "-2"},
+                {"Boolean", "TRUe"},
+                {"Nested:Integer", "11"}
+            };
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dic);
+            var config = configurationBuilder.Build();
+
+            var instance = new ComplexOptions();
+
+            Assert.Throws<InvalidOperationException>(
```

Comment:
perhaps you can get the exception and assert the message as well:

Something similar to:

https://github.com/dotnet/runtime/blob/342e4cd67b1337308a5f7d281c882dcc93c1b29c/src/libraries/Microsoft.Extensions.Configuration/tests/ConfigurationTest.cs#L539-L543

Reply:
That makes sense, thank you.  In my day job, I generally don't test exception messages.  I don't do this because we don't use string resources which means that if someone tweaks the message, it'll needlessly break the test. I do however test 'fragments' of the message, e.g. I'd want to make sure it `contains` the property names I'm expecting.